spec/factories/ect_at_school_period_factory.rb

```diff
@@ -4,10 +4,14 @@
   factory(:ect_at_school_period) do
     association :school
     association :teacher
+    association :lead_provider
+    association :appropriate_body
```

Comment:
`lead_provider` and `appropriate_body` are not required for an ECTAtSchoolPeriod, so I wouldn't create them by default but only explicitly in the spec that needs it.